api/src/services/applications/create_application.py

```diff
@@ -18,8 +19,36 @@
 logger = logging.getLogger(__name__)
 
 
+def _validate_organization_membership(
+    db_session: db.Session, organization_id: UUID, user: User
+) -> Organization:
+    # Fetch the organization
+    organization = db_session.execute(
+        select(Organization).where(Organization.organization_id == organization_id)
+    ).scalar_one_or_none()
+
+    if not organization:
+        raise_flask_error(404, "Organization not found")
+
+    # Check if the user is a member of the organization
+    is_member = db_session.execute(
+        select(OrganizationUser)
+        .where(OrganizationUser.organization_id == organization_id)
+        .where(OrganizationUser.user_id == user.user_id)
+    ).scalar_one_or_none()
```

Comment:
I think if we add a selectinload for the organization user, then the second query here wouldn't even have to hit the DB as we'd already have the values cached. Very minor optimizations, but quick to switch.

```suggestion
    organization = db_session.execute(
        select(Organization).where(Organization.organization_id == organization_id).options(selectinload(Organization.organization_users))
    ).scalar_one_or_none()

    if not organization:
        raise_flask_error(404, "Organization not found")

    # Check if the user is a member of the organization
    is_member = db_session.execute(
        select(OrganizationUser)
        .where(OrganizationUser.organization_id == organization_id)
        .where(OrganizationUser.user_id == user.user_id)
    ).scalar_one_or_none()
```